File: lib/github/repository.rb
# frozen_string_literal: true

module GitHub
  Repository = Data.define(
    :name,                 #: String
    :url,                  #: String
    :pull_requests_count,  #: Integer
    :language_versions,    #: Hash[String, Array[String]]
    :no_actionlint         #: bool
  )
end

File: src/github/repository.ts
export interface PullRequest {
  title: string;
  url: string;
  labels: string[];
}

export interface Repository {
  name: string;
  url: string;
  pullRequests: PullRequest[];
  languageVersions: Record<string, string[]>;
  noActionlint: boolean;
  noDependabot: boolean;
  noDependabotCooldown: boolean;
}
